odbc/src/api/data.rs

```diff
@@ -1104,7 +1125,12 @@ pub mod i16_len {
         // size_of::<WideChar>() will always be 2, so this cast is safe
         #[allow(clippy::cast_possible_truncation, clippy::cast_possible_wrap)]
         // Only copy the length if the pointer is not null
-        ptr_safe_write(text_length_ptr, (size_of::<WideChar>() * len) as SmallInt);
+        ptr_safe_write(
+            text_length_ptr,
+            (size_of::<WideChar>() * len)
+                .try_into()
+                .unwrap_or(SmallInt::MAX),
```

Comment:
I don't like copying a default value. If text_length_ptr points is a pointer to a type too small to hold the value, then we should fail.
It should never happen. But, just in case, if it does, it is easier to notice it when there is a failure than when the returned value is silently assigned some default.